### ISSUE TITLE
Support for partials

### ISSUE THREAD

**Issue by rowild** [WindsonFF/vite-plugin-virtual-html#7]
Is there any support for any kind of partial system like ejs, handlebars, pug, moustache etc?

**Comment by WindsonFF** [comment]
you can try 0.1.1

i've add basic support for this.

**Comment by rowild** [comment]
@WindsonFF Thank you for working on that! :-)

However, I tried it and it does not work. Here is a codesandbox:
https://codesandbox.io/s/vite-with-virtual-html-plugin-92pkx

I experience a series of problems, maybe you find a solution for that. E.g.:
- I have no idea how to call the url to "subpage/index.html". This can be seen in the "root" index.html, which tries to link to the subpage/index.html.
- the "root" index tries to include a partial, but without success. How does the system calculate the path to the partial? From the current template the partial is referenced from? Or from a pre-defined folder (a "source path")? If it is the latter, where is that "source path" defined?

Any ideas what I am doing wrong?

**Comment by WindsonFF** [comment]
@rowild looks like it is not an error of this plugin.maybe it's your 'code' not good.

i'v test this and look `ejs` doc, i found that maybe you have two errors.

1.  in  `vite.config` file , maybe you need to add this to `ejs.render` 's 3rd param. `root: process.cwd()`. to use the absolute file path
2. in `index.html` file, maybe you should use `<$= include('/demo/partials/test') $>` to replace this code `<$= include /src/partials/text.html $>` . i didn't found this code format in `ejs` docs.

**Comment by rowild** [comment]
@WindsonFF Thank you for your quick feedback again! :-)

I am sed to the webpack way of including ejs partials, so I haven't thought about the possibility to use real ejs. Stupid me!
However, according to the ejs documentation `a partial is located relative to the file where it is included`. I tried that in my codesandbox example, but it still does not work.

i do not understand what you mean by adding `root: process.cwd()` to the 3rd paramter of the `ejs.ender` function. The 3rd parameter defines the delimiter... and I did not find anything about that in the ejs docu. Any chance you could "fix" my codesandbox example? That would be the best way to show how it's done correctly. Thank you!

Also, there is still the question of how to create a link to the subpage or even simply call the subpage. What is the proper URL to the subpage in a setup that does not use a router like Vue does?

**Comment by zoranpopovic** [comment]
It is a devDependency, which won't get installed if you just include vite-plugin-virtual-html in your project's package.json.  I think that's fine, because it is an optional dependency.  For clarity, the readme should point out that if templates are used, the render() option needs to be provided for the plugin, and that it's not enough to just have {html, data} for provided.

**Comment by zoranpopovic** [comment]
@rowild i think the author was referring to the third parameter of the render function (e.g. render(template,data,thirdParam)) which is an object...

**Comment by zoranpopovic** [comment]
example is too involved to share, but here's the relevant part of the vite.config.js
```
const pages = {};
glob.sync(resolve(__dirname, "Projects/Activities/ClientApp/Pages/*.js")).forEach(file => {
  const name = basename(file, ".js");

  pages[name] = {
    html: "/ViteTemplate.html",
    data: { name },
  };
});


export default defineConfig({
  base: "/",
  plugins: [
    createVuePlugin(/*options*/),
    virtualHtml({
      pages,
      indexPage: "Home",
      render(template, data) {
        return render(template, data, { root: process.cwd() });
      },
    }),
```

it basically sets up a page or each page in Projects/Activities/ClientApp/Pages/ directory using a basic template in ViteTemplate.html that takes a single parameter `name`

**Comment by WindsonFF** [comment]
> @zoranpopovic Thank you! Those parts I got working. (And I just saw that I already import ejs, calling `ejs..render(...)` eventually.
> 
> But the problem are partials, which are not recognised. And a solution how to make a navigation. None of that works in my codesandbox example (to which I added the `root: process.cwd()`): https://codesandbox.io/s/vite-with-virtual-html-plugin-92pkx
> 
> Did you get these things running?
> 
> UPDATE: I think I should once more point out that I am looking for a partial solution without any of those frameworks like Vue or React. Reason being that I have to create click dummies that will eventually be integrated into a CMS that has its own templating framework. Which means the resulting CSS and JS should be free of those frameworks and the components should be mirrored from what is created within vite with a template partial system.

did you still have `partial file` error?
how about look this: 
https://github.com/WindsonFF/vite-plugin-virtual-html/blob/d0b747aa1158ec85f2b147ae4171c30c5154b523/vite.config.ts#L30
and
https://github.com/WindsonFF/vite-plugin-virtual-html/blob/d0b747aa1158ec85f2b147ae4171c30c5154b523/demo/demo1/demo1.html#L11

**Comment by rowild** [comment]
@WindsonFF So as it turns out there are 2 things: 

1. those who can read and write, clearly have a better life. I called my template "test", but required "teXt". SHAME ON ME!
2. The so-called "relative" path seems to be relative form `vite.config.js`, NOT from the file where the partial is included. This is actually NOT relative in my understanding, but absolute from the root directory.

So thank you for your help and patience - I got that part working!

Remains the question how to build a navigation? If I can ask you once more to have a look at my codesandbox example you will see that there is a link to a subpage. But no matter how I formulate the link (`/src/pages/subpage/index.html`, `./subpage/index.html` ...) it won't work. 

How can links like these be resolved without a router?
